CONTRIBUTING.md

```diff
@@ -2,27 +2,115 @@
+work, under the same license.  Copyright and license notices must be preserved.  Contributors provide
+an express grant of patent rights.
+
+If you are willing to follow the conditions of this license, check out the following links
+to find an unassigned item to contribute to:
+
+- ["Good first issue" GitHub issues for first-timers](https://github.com/orcasound/orcasite/labels/good%20first%20issue)
+- ["Help wanted" GitHub issues](https://github.com/orcasound/orcasite/labels/help%20wanted)
+- [Project board](https://github.com/orgs/orcasound/projects/38/views/1)
+- [Figma](https://www.figma.com/design/41JuvNkXJhTUob8HMsJiNC/Orcasite%3A-design-updates?node-id=882-3971)
+- [Trello board with overview of features](https://trello.com/b/hRFh7Sc1/orcasite-development)
+- [Public roadmap](https://trello.com/b/wBg0qhss/orcasound-roadmap)
+
+Once you've chosen an issue and want to contribute code:
 
-0. Check out the [Trello board](https://trello.com/b/hRFh7Sc1/orcasite-development) for an overview of features being worked on.
-    - This [public roadmap](https://trello.com/b/wBg0qhss/orcasound-roadmap) has a higher level view.
 1. Fork the Orcasite repo.
 2. Develop on a [feature branch](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow).
 3. Submit a PR (don't review your own)!
     - To maintain a consistent style, we recommend running [Prettier](https://github.com/prettier/prettier) on js, and `mix format` before submission.
 4. Once a PR is merged, we can throw it up onto the dev server to see how things look!
 
+### Submitting a Pull Request
+
+For all but the absolute simplest changes, first find an existing GitHub issue or
+[submit a new issue](https://github.com/orcasound/orcasite/issues/new) so that the
+community can review and provide feedback. Getting early feedback will help ensure your work
+is accepted by the community. This will also allow us to better coordinate our efforts and
+minimize duplicated effort.
+
+If you would like to contribute, first identify the scale of what you would like to contribute.
+If it is small (grammar/spelling or a bug fix) feel free to start working on a fix. If you are
+submitting a feature or substantial code contribution, please discuss it with the maintainers and
+ensure it follows the public roadmap. You might also read these two blogs posts on contributing
+code: [Open Source Contribution Etiquette](http://tirania.org/blog/archive/2010/Dec-31.html) by Miguel de Icaza and
+[Don't "Push" Your Pull Requests](https://www.igvita.com/2011/12/19/dont-push-your-pull-requests/) by Ilya Grigorik.
+All code submissions will be reviewed and tested, and only those that meet
+the bar for both quality and design/roadmap appropriateness will be merged into the source.
+
+For all pull requests the following rules apply:
+- The pull request description should describe the problem and solution, and reference the GitHub issue if one exists.
```

Comment:
I can find existing PRs like https://github.com/orcasound/orcasite/pull/635 that don't follow this rule.
However, I still recommend that this rule should apply.